lib/keys/calendarKeys.ts

```diff
@@ -0,0 +1,196 @@
+
+import { ENCRYPTION_TYPES, ENCRYPTION_CONFIGS } from '../constants';
+import { normalize } from '../helpers/string';
+import { serializeUint8Array } from '../helpers/serialization';
+import { hasBit } from '../helpers/bitset';
+import { Address, EncryptionConfig } from '../interfaces';
+import { Key as tsKey, Member } from '../interfaces/calendar';
+
+export const KEY_FLAGS = {
+    PRIMARY: 0b1,
+    ACTIVE: 0b01
+};
+
+export const generatePassphrase = () => {
+    const value = getRandomValues(new Uint8Array(32));
+    return serializeUint8Array(value);
+};
+
+/**
+ * The calendar key is generated with less user info to not confuse if the key is exported.
+ */
+export const generateCalendarKey = async ({
+    passphrase,
+    encryptionConfig
+}: {
+    passphrase: string;
+    encryptionConfig: EncryptionConfig;
+}) => {
+    const { key: privateKey, privateKeyArmored } = await generateKey({
+        userIds: [{ name: 'Calendar key' }],
+        passphrase,
+        ...encryptionConfig
+    });
+
+    await privateKey.decrypt(passphrase);
+
+    return { privateKey, privateKeyArmored };
+};
+
+export const encryptPassphrase = async ({
+    passphrase,
+    privateKey,
+    memberPublicKeys
+}: {
+    passphrase: string;
+    privateKey: OpenPGPKey;
+    memberPublicKeys: { [key: string]: OpenPGPKey };
+}) => {
+    const memberPublicKeysList = Object.entries(memberPublicKeys);
+    const { data, signature } = await encryptMessage({
+        message: await createMessage(passphrase),
+        publicKeys: memberPublicKeysList.map(([, publicKey]) => publicKey),
+        privateKeys: [privateKey],
+        detached: true
+    });
+    const message = await getMessage(data);
+    const { asymmetric, encrypted } = await splitMessage(message);
+
+    return {
+        keyPackets: memberPublicKeysList.reduce((acc, [memberID], index) => {
+            acc[memberID] = serializeUint8Array(asymmetric[index]);
+            return acc;
+        }, Object.create(null)),
+        dataPacket: serializeUint8Array(encrypted[0]),
+        signature
+    };
+};
+
+/**
+ * Decrypts a calendar passphrase with a private key
+ */
+export const decryptPassphrase = async ({
+    armoredPassphrase,
+    armoredSignature,
+    privateKeys,
+    publicKeys
+}: {
+    armoredPassphrase: string;
+    armoredSignature: string;
+    privateKeys: OpenPGPKey[];
+    publicKeys: OpenPGPKey[];
+}) => {
+    const { data: decryptedPassphrase, verified } = await decryptMessage({
+        message: await getMessage(armoredPassphrase),
+        signature: await getSignature(armoredSignature),
+        privateKeys,
+        publicKeys
+    });
+
+    if (verified !== VERIFICATION_STATUS.SIGNED_AND_VALID) {
+        const error = new Error(c('Error').t`Signature verification failed`);
+        error.name = 'SignatureError';
+        throw error;
+    }
+
+    return decryptedPassphrase;
+};
+
+export const getAddressesMembersMap = (Members: Member[] = [], Addresses = []) => {
```

Comment:
Probably doesn't really matter here, but in general it's awkward to put optional params in front because `fn(undefined, someValue)` vs `fn(someValue)`.
Probably you just added types here anyway.